core/pkg/resultshandling/printer/v2/sarifprinter.go

```diff
@@ -110,57 +111,81 @@ func (sp *SARIFPrinter) addResult(scanRun *sarif.Run, ctl reportsummary.IControl
 		})
 }
 
-func (sp *SARIFPrinter) PrintImageScan(context.Context, *models.PresenterConfig) {
+func (sp *SARIFPrinter) PrintImageScan(scanResults *models.PresenterConfig) error {
+	if scanResults == nil {
+		return fmt.Errorf("no no image vulnerability data provided")
+	}
+
+	presenterConfig, err := presenter.ValidatedConfig("sarif", "", false)
+	if err != nil {
+		return err
+	}
+
+	pres := presenter.GetPresenter(presenterConfig, *scanResults)
+
+	return pres.Present(sp.writer)
 }
 
 func (sp *SARIFPrinter) PrintNextSteps() {
 
 }
 
 func (sp *SARIFPrinter) ActionPrint(ctx context.Context, opaSessionObj *cautils.OPASessionObj, imageScanData []cautils.ImageScanData) {
-	report, err := sarif.New(sarif.Version210)
-	if err != nil {
-		panic(err)
-	}
+	if opaSessionObj == nil {
+		if len(imageScanData) > 0 {
+			if err := sp.PrintImageScan(imageScanData[0].PresenterConfig); err != nil {
+				logger.L().Ctx(ctx).Error("failed to write results in sarif format", helpers.Error(err))
+				return
+			}
+		} else {
+			logger.L().Ctx(ctx).Fatal("failed to write results in sarif format: no data provided")
+		}
 
-	run := sarif.NewRunWithInformationURI(toolName, toolInfoURI)
-	basePath := getBasePathFromMetadata(*opaSessionObj)
+	} else {
```

Comment:
Why `else`? You should end the function if you entered to `if` statement

Reply:
I have used `else` since at the end of the function there is a call to `printer.LogOutputFile(sp.writer.Name())` which is used by both flows